### PR TITLE
Use stat gem

### PR DESCRIPTION
**Pull request by ilyalyo**
close https://github.com/mcandre/enlint/issues/12

### CODE REVIEW

File: lib/enlint.rb [hunk 1]
```diff
@@ -151,23 +151,14 @@ def to_finding(encoding_difference = false)
       preferred = encoding_difference[1].inspect
 
       if observed == NO_SUCH_FILE
-        "#{@filename}: #{NO_SUCH_FILE}"
+        finding = StatModule::Finding.new(false, 'File doesn\'t exist', "#{@filename}: #{NO_SUCH_FILE}")
```
Comment (mcandre) [inline lib/enlint.rb]: `File does not exist`

Avoids escape codes, helps ESL users

File: lib/enlint.rb [hunk 2]
```diff
@@ -151,23 +151,14 @@ def to_finding(encoding_difference = false)
...
-                "Compatibility"
-            ],
-            :location => {
-                :path => "#{@filename}",
-            },
-            :recommendation => "Prefered #{preferred}"
-        }
+        finding = StatModule::Finding.new(true, 'File encoding', "Observed #{observed}")
+        finding.categories = ['Compatibility']
+        finding.location = StatModule::Location.new("#{@filename}")
```
Comment (mcandre) [inline lib/enlint.rb]: Again, `@filename.to_s`

File: lib/enlint.rb (hunk 1 above)
Comment (ilyalyo) [inline lib/enlint.rb]: Got you, fixed.

File: lib/enlint.rb (hunk 2 above)
Comment (ilyalyo) [inline lib/enlint.rb]: fixed

**Comment by ilyalyo** [comment]
This PR is fixed.

**Pull request merged**